core/vision/src/main/java/com/arcrobotics/ftclib/vision/UGBasicHighGoalPipeline.java

```diff
@@ -51,13 +52,25 @@ public UGBasicHighGoalPipeline() {
         redRect = new Rect();
 
         minThreshold = 155;
-        maxThreshold = 255;
+        maxThreshold = 200;
     }
 
     @Override
-    public Mat processFrame(Mat input) {
+    public void init(Mat mat) {
+        super.init(mat);
+        int imageWidth = mat.width();
+        int imageHeight = mat.height();
+
+        centerX = ((double) imageWidth / 2) - 0.5;
+        centerY = ((double) imageHeight / 2) - 0.5;
+    }
 
+    public boolean filterContours(MatOfPoint contour) {
+        return Imgproc.contourArea(contour) > 30;
+    }
```

Comment:
Does this need to be public?

Reply:
Also, can we make the `30` modifiable? I feel like that may be subject to different constraints/conditions.